security-profile/security_profile.go

```diff
@@ -84,8 +84,39 @@ func (p *Profile) AddNamespaces(nsTypes ...specs.LinuxNamespaceType) {
 	}
 }
 
+/* Add seccomp rules to allow syscalls with the given arguments if necessary */
+func (p *Profile) AllowSyscallsWithArgs(syscallsWithArgsToAllow map[string][]specs.LinuxSeccompArg) {
+	defaultActError := (p.Oci.Linux.Seccomp.DefaultAction == specs.ActErrno)
+
+	for syscallNameToAllow, syscallArgsToAllow := range syscallsWithArgsToAllow {
+
+		for _, syscallRule := range p.Oci.Linux.Seccomp.Syscalls {
+
+			if syscallRule.Action == specs.ActAllow {
+				for _, syscallName := range syscallRule.Names {
+					if syscallName == syscallNameToAllow &&
+						((len(syscallArgsToAllow) == 0 && len(syscallRule.Args) == 0) ||
+							reflect.DeepEqual(syscallRule.Args, syscallArgsToAllow)) {
+						return
```

Comment:
should this be `continue`?  This function is hard to follow so I might be misunderstanding